src/scheduler/node_info.c

```diff
@@ -3292,6 +3324,85 @@ is_vnode_eligible_chunk(resource_req *specreq, node_info *node,
 
 /**
  * @brief
+ *	Checks if a vnode is eligible for power operations.
+ *  Based on is_provisionable.
+ *
+ * @par Functionality:
+ *	This function checks if a vnode is eligible to be provisioned.
+ *	A vnode is eligible for power operations if it satisfies all of the
+ *	following conditions:-
+ *	(1) Server has power_provisioning True,
+ *	(2) Vnode has power_provisioning True,
+ *	(3) No conflicts with reservations already running on the Vnode
+ *	(4) No conflicts with jobs already running on the Vnode
+ *
+ * @param[in]		node	-	pointer to node_info
+ * @param[in]		resresv	-	pointer to resource_resv
+ * @param[in,out]	err		-	pointer to schd_error
+ *
+ * @return	int
+ * @retval	 NO_PROVISIONING_NEEDED : resresv doesn't request eoe
+ *			or resresv is not a job
+ * @retval	 PROVISIONING_NEEDED : vnode doesn't have current_eoe set
+ *          or it doesn't match job eoe
+ * @retval	 NOT_PROVISIONABLE  : vnode is not provisionable
+ *			(see err for more details)
+ *
+ * @par Side Effects:
+ *	Unknown
+ *
+ * @par MT-safe: No
+ *
+ */
+int
+is_powerok(node_info *node, resource_resv *resresv, schd_error *err)
+{
+	int i;
+	int ret = NO_PROVISIONING_NEEDED;
+
+	if (!resresv->is_job)
+		return NO_PROVISIONING_NEEDED;
+	if (resresv->eoename == NULL)
+		return NO_PROVISIONING_NEEDED;
+	if (!resresv->server->power_provisioning) {
+		err->error_code = PROV_DISABLE_ON_SERVER;
+		return NOT_PROVISIONABLE;
+	}
+
+	/* node doesn't have eoe or it doesn't match job eoe */
+	if (node->current_eoe == NULL ||
+			strcmp(resresv->eoename, node->current_eoe) != 0) {
+		ret = PROVISIONING_NEEDED;
+
+		if (!node->power_provisioning) {
```

Comment:
can't this be moved out of the condition? If the node itself does not support power provisioning then why do forward.